### PR TITLE
feat: Ktor implementation for auth service

### PR DESCRIPTION
**Pull request by PratyushSingh07**

### CODE REVIEW

File: core/network/build.gradle.kts [hunk 1]
```diff
@@ -61,6 +61,8 @@ dependencies {
     implementation("io.ktor:ktor-client-json:2.3.4")
     implementation("io.ktor:ktor-client-websockets:2.3.4")
     implementation("io.ktor:ktor-serialization-kotlinx-json:2.3.4")
+    implementation("org.jetbrains.kotlinx:kotlinx-serialization-json:1.3.0")
```
Comment (therajanmaurya) [inline core/network/build.gradle.kts]: We have serialization plugin in model module and model module should only contains network models. 

Why did you add the serialization in network module?

File: core/network/src/main/kotlin/org/mifospay/core/network/di/NetworkModule.kt [hunk 2]
```diff
@@ -177,8 +177,8 @@ class NetworkModule {
 
     @Provides
     @Singleton
-    fun provideAuthenticationService(client: HttpClient): KtorAuthenticationService {
-        return KtorAuthenticationService(client)
+    fun provideAuthenticationService(client: HttpClient, preferencesHelper: PreferencesHelper): KtorAuthenticationService {
```
Comment (therajanmaurya) [inline core/network/src/main/kotlin/org/mifospay/core/network/di/NetworkModule.kt]: Don't we have the datastore? @AdityaKumdale recently migrated the SharedPrefrence with Datastore.

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: core/network/build.gradle.kts (hunk 1 above)
Comment (PratyushSingh07) [reply core/network/build.gradle.kts]: this is done

File: core/network/src/main/kotlin/org/mifospay/core/network/di/NetworkModule.kt (hunk 2 above)
Comment (PratyushSingh07) [reply core/network/src/main/kotlin/org/mifospay/core/network/di/NetworkModule.kt]: will have to crosscheck this with aditya

Comment (AdityaKumdale) [reply core/network/src/main/kotlin/org/mifospay/core/network/di/NetworkModule.kt]: Yes we have implemented proto datastore instead of preference store

Comment (therajanmaurya) [reply core/network/src/main/kotlin/org/mifospay/core/network/di/NetworkModule.kt]: I don't see any implementation of datastore in `datastore` module @AdityaKumdale why?
